src/v/raft/state_machine_manager.cc

```diff
@@ -431,7 +431,9 @@ ss::future<> state_machine_manager::background_apply_fiber(
   entry_ptr entry, ssx::semaphore_units units) {
     while (!_as.abort_requested() && entry->stm->next() < _next) {
         storage::log_reader_config config(
```

Comment:
I'm curious if you checked other calls of this `storage::log_reader_config` constructor? Since max is inclusive there is potential for more off-by-1 problems.

Reply:
i did check it in `state_machine_manager` and they are fine. Will audit the other parts of the code